### PR TITLE
Send conversion method suggestions

### LINKED ISSUE #7320: Ability to get the list of target conversion types for a type
In order to allow use of [Any.to](https://github.com/enso-org/enso/pull/7704) conversion method easier and conversions more powerful we want to enable **finding a list of target types for a value**. For example: `Column` will be convertible to `Vector`, `Text`, `JSON`, `Table`.

##### Component Browser

The IDE can then display/offer not only `Column` own methods in the **component browser**, but also methods on the _convertible to types_ when having a _value of type_ `Column`. Making it possible to use a value in _all its possible roles_ without requiring the user to do such conversions in multiple steps.

##### Example

If there is a node with value of type `Complex` (just like in [this example](https://github.com/enso-org/enso/pull/7716#issuecomment-1704822519)) and there is a conversion from `Complex` to `Number`, then the **component browser** may offer method like `Number.floor`  and when it is selected, it would insert:
```ruby
complexOperator1.to Number . floor
```
e.g. it would convert the value of type `Complex` to `Number` via `Any.to` method (which effectively invokes the `Number.from(that:Complex)` method and then operated on the result as it was a `Number`. 
```[tasklist]
### Tasks
- [ ] Enable transfer of conversion methods (current it is [disabled](https://github.com/enso-org/enso/pull/7716/files#diff-a371d887fb5a47107c1310b0d3707d3505f2dd28e96b1832e1b6041fcaed2890R749))
- [ ] Cleanup the DB/send data (here is the [list of current problems](https://github.com/enso-org/enso/issues/7320#issuecomment-1704972767))
- [ ] Make sure IDE receives and stores the data
- [ ] Demonstrate the IDE can use the data (showing more methods for example when `Complex` value is present)
```

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

close #7320

Changelog:
- update: enable conversion suggestions
- fix: conversion suggestion building
- fix: conversion suggestion types
- fix: conversion JSON-RPC representation

### Important Notes

For example, the [`Day_Of_Week_From`](https://github.com/enso-org/enso/blob/5150c14afde5af0cea3b6ec747576952b86d2160/distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Day_Of_Week_From.enso) conversion is sent as

```json
{
   "type":"Add",
   "id":32,
   "suggestion":{
      "type":"method",
      "module":"Standard.Base.Data.Time.Day_Of_Week_From",
      "name":"from",
      "arguments":[
         {
            "name":"that",
            "reprType":"Standard.Base.Data.Numbers.Integer",
            "isSuspended":false,
            "hasDefault":false,
            "defaultValue":null,
            "tagValues":null
         },
         {
            "name":"first_day",
            "reprType":"Standard.Base.Data.Time.Day_Of_Week.Day_Of_Week",
            "isSuspended":false,
            "hasDefault":true,
            "defaultValue":"Day_Of_Week.Sunday",
            "tagValues":[
               "Standard.Base.Data.Time.Day_Of_Week.Day_Of_Week.Sunday",
               "Standard.Base.Data.Time.Day_Of_Week.Day_Of_Week.Monday",
               "Standard.Base.Data.Time.Day_Of_Week.Day_Of_Week.Tuesday",
               "Standard.Base.Data.Time.Day_Of_Week.Day_Of_Week.Wednesday",
               "Standard.Base.Data.Time.Day_Of_Week.Day_Of_Week.Thursday",
               "Standard.Base.Data.Time.Day_Of_Week.Day_Of_Week.Friday",
               "Standard.Base.Data.Time.Day_Of_Week.Day_Of_Week.Saturday"
            ]
         },
         {
            "name":"start_at_zero",
            "reprType":"Standard.Base.Data.Boolean.Boolean",
            "isSuspended":false,
            "hasDefault":true,
            "defaultValue":"False",
            "tagValues":[
               "Standard.Base.Data.Boolean.Boolean.True",
               "Standard.Base.Data.Boolean.Boolean.False"
            ]
         }
      ],
      "selfType":"Standard.Base.Data.Time.Day_Of_Week.Day_Of_Week",
      "returnType":"Standard.Base.Data.Time.Day_Of_Week.Day_Of_Week",
      "isStatic":false,
      "documentation":" Convert from an integer to a Day_Of_Week\n\nArguments:\n- `that`: The first day of the week.\n- `first_day`: The first day of the week.\n- `start_at_zero`: If True, first day of the week is 0 otherwise is 1.",
      "annotations":[
         
      ]
   }
}
```

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.

### CODE REVIEW

**Comment by JaroslavTulach** [comment]
> For example, the [`Day_Of_Week_From`](https://github.com/enso-org/enso/blob/5150c14afde5af0cea3b6ec747576952b86d2160/distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Day_Of_Week_From.enso) conversion is sent as

Thank you @4e6, I think the information sent shall be sufficient for the IDE to _find out all "roles" a "value at hand" can play (be converted to)_:

>       "name":"from",

The IDE can search for *all methods named* `from`...

>       "arguments":[
>          {
>             "name":"that",
>             "reprType":"Standard.Base.Data.Numbers.Integer",
>             "isSuspended":false,
>             "hasDefault":false,
>             "defaultValue":null,
>             "tagValues":null
>          },

...where the `that` argument has the `reprType` of the _"value at hand"_ ...

>       "returnType":"Standard.Base.Data.Time.Day_Of_Week.Day_Of_Week",

...and include the methods of `returnType` in _component browser_ for such _"value at hand"_. Once such a method is selected, the IDE would insert `value_at_hand.to Day_Of_Week . method_of_day_of_week`.

File: engine/runtime/src/test/scala/org/enso/compiler/test/context/SuggestionBuilderTest.scala [hunk 1]
```diff
@@ -908,10 +919,16 @@ class SuggestionBuilderTest extends AnyWordSpecLike with Matchers {
...
+                  "Standard.Base.Data.Numbers.Number",
+                  false,
+                  false,
+                  None
+                )
               ),
-              returnType    = "Unnamed.Test.MyType",
-              selfType      = "Number",
+              selfType      = "Standard.Base.Data.Numbers.Number",
+              returnType    = "Unnamed.Test.Foo",
```
Comment (JaroslavTulach) [inline engine/runtime/src/test/scala/org/enso/compiler/test/context/SuggestionBuilderTest.scala:931]: Yes, that seems correct. `selfType` is now FQN. `returnType` is the type we convert to.

File: engine/runtime/src/test/scala/org/enso/compiler/test/context/SuggestionBuilderTest.scala [hunk 2]
```diff
@@ -1046,12 +1124,31 @@ class SuggestionBuilderTest extends AnyWordSpecLike with Matchers {
...
+                  "that",
+                  "Standard.Base.Data.Numbers.Number",
+                  false,
+                  false,
+                  None
+                ),
                 Suggestion
-                  .Argument("opt", "Unnamed.Test.MyMaybe", false, false, None)
+                  .Argument(
+                    "other",
```
Comment (JaroslavTulach) [inline engine/runtime/src/test/scala/org/enso/compiler/test/context/SuggestionBuilderTest.scala:1136]: Optional arguments are available as well. Good.

**Review by JaroslavTulach — APPROVED** [review]

**Review by hubertp — COMMENTED** [review]
:+1:

**Pull request merged**